packages/ledger-icrc/src/canister.ts

```diff
@@ -0,0 +1,23 @@
+import { Canister, toNullable } from "@dfinity/utils";
+import type { _SERVICE as IcrcIndexService } from "../candid/icrc_index";
+import type {
+  _SERVICE as IcrcLedgerService,
+  Tokens,
+} from "../candid/icrc_ledger";
+import type { BalanceParams } from "./types/ledger.params";
+
+export abstract class IcrcCanister<
```

Comment:
Is this necessary? Will they share more methods in the future?

Reply:
Not sure, but given that it used a feature called `icrc1_balance_of` and other features are prefixed with `icrc1_` in the ledger, I assumed there might be more in the future. Moreover, at the time we integrated this, I think there were discussions about replicating more endpoints. It also avoids duplication of code.

Therefore, I would say it's not a must, but it's a nice feature.